internal/server/iigointernal/orchestration.go

```diff
@@ -74,10 +74,25 @@ func RunIIGO(g *gamestate.GameState, clientMap *map[shared.ClientID]baseclient.C
 	iigoClients = *clientMap
 
 	// Increments the budget by a constant 100
-	// TODO:- the constant should be retrieved from the rules
-	g.IIGORolesBudget[shared.President] += 100
-	g.IIGORolesBudget[shared.Judge] += 100
-	g.IIGORolesBudget[shared.Speaker] += 100
+	PresidentIncRule := rules.AvailableRules["increment_budget_president"].ApplicableMatrix
+	PresidentIncrement := PresidentIncRule.At(0, 1)
+	g.IIGORolesBudget[shared.President] += shared.Resources(PresidentIncrement)
+	JudgeIncRule := rules.AvailableRules["increment_budget_judge"].ApplicableMatrix
+	JudgeIncrement := JudgeIncRule.At(0, 1)
+	g.IIGORolesBudget[shared.Judge] += shared.Resources(JudgeIncrement)
+	SpeakerIncRule := rules.AvailableRules["increment_budget_speaker"].ApplicableMatrix
+	SpeakerIncrement := SpeakerIncRule.At(0, 1)
+	g.IIGORolesBudget[shared.Speaker] += shared.Resources(SpeakerIncrement)
```

Comment:
Should you check rulesInPlay instead of available rules and only increment if it is in the rule is in play?

Reply:
oh right forgot that part